src/components/dock/dock-button/dock-button.tsx

```diff
@@ -69,6 +78,18 @@ export class DockButton {
         return this.renderButton(this.handleClick);
     }
 
+    @Watch('isOpen')
+    protected openWatcher() {
+        if (!this.isOpen) {
```

Comment:
it works like a charm! Great job!

My only comment is about the interaction design. After the menu is opened, if user press <kbd>esc</kbd>, the focus will not go back to the trigger element. 

Not sure if it's possible to fix, but it'd be great if user could close the menu using the keyboard, and continue their navigation pattern. This could however be fixed in another PR too.